store/views.py

```diff
@@ -12,21 +16,61 @@ def index(request):
 
 def bookDetailView(request, bid):
     template_name = 'store/book_detail.html'
+    book1=Book.objects.get(pk=bid)
+    num_available=BookCopy.objects.filter(status__exact=True,book__exact=book1).count()
+    user_rating=0.0
+    if(request.user.is_authenticated):
+        value= BookRating.objects.filter(book__exact=book1,user__exact=request.user)
+        user_rating=0
+        if(value.count()>0):
+            user_rating=BookRating.objects.filter(book__exact=book1,user__exact=request.user).get().ratinguser
+
     context = {
-        'book': None, # set this to an instance of the required book
-        'num_available': None, # set this to the number of copies of the book available, or 0 if the book isn't available
+        'book': book1, # set this to an instance of the required book
+        'num_available':num_available, # set this to the number of copies of the book available, or 0 if the book isn't available
+        'user_rating':user_rating,
     }
-    # START YOUR CODE HERE
-    
-    
     return render(request, template_name, context=context)
 
+@csrf_exempt
+def bookRatingView(request):
+    template_name= template_name = 'store/book_detail.html'
+    book_id = request.POST.get("bid")
+    rating_value = request.POST.get("rating")
```

Comment:
You've not put a backend validation on the rating, so the user can simply edit the JS code you've written in the template and easily put invalid values of rating.